asset/js/widget/BaseInput.js

```diff
@@ -666,77 +688,95 @@ define(["../notjQuery", "Completer"], function ($, Completer) {
+                                    this.clearPartialTerm(input);
+                                } else {
+                                    this.writePartialTerm(input.value.slice(0, -1), input);
+                                }
+
+                                removedTerms[termIndex] = removedTerm;
+                                event.preventDefault();
                             }
-
-                            removedTerms[termIndex] = removedTerm;
-                            event.preventDefault();
-                        }
-                    } else if (isNaN(termIndex)) {
-                        if (! input.value && this.hasTerms()) {
-                            let termData = this.popTerm();
-                            if (! event.ctrlKey && ! event.metaKey) {
-                                // Removing the last char programmatically is not
-                                // necessary since the browser default is not prevented
-                                this.writePartialTerm(termData.label, input);
+                        } else if (isNaN(termIndex)) {
+                            if (! input.value && this.hasTerms()) {
+                                let termData = this.popTerm();
+                                if (! event.ctrlKey && ! event.metaKey) {
+                                    // Removing the last char programmatically is not
+                                    // necessary since the browser default is not prevented
+                                    this.writePartialTerm(termData.label, input);
+                                }
+
+                                removedTerms[this.usedTerms.length] = termData;
                             }
-
-                            removedTerms[this.usedTerms.length] = termData;
                         }
-                    }
 
-                    this.togglePlaceholder();
-                    this.autoSubmit(input, 'remove', removedTerms);
+                        this.togglePlaceholder();
+                        this.autoSubmit(input, 'remove', removedTerms);
+                    }
                     break;
                 case 'Delete':
-                    removedTerms = this.clearSelectedTerms();
-
-                    if (termIndex >= 0 && ! input.value) {
-                        let removedTerm = this.removeTerm(input.parentNode);
-                        if (removedTerm !== false) {
-                            input = this.moveFocusForward(termIndex - 1);
-                            if (event.ctrlKey || event.metaKey) {
-                                this.clearPartialTerm(input);
-                            } else {
-                                this.writePartialTerm(input.value.slice(1), input);
+                        removedTerms = this.clearSelectedTerms();
+
+                        if (termIndex >= 0 && (! input.value || this.isTermDirectionVertical())) {
```

Comment:
Isn't this about the read-only mode? Or why should the value be not of interest just because the direction is vertical?

Reply:
Oh, I suggested it myself... This **is** rather about the read-only mode. I see no reason why `Delete` should be handled at all if `Backspace` isn't handled in vertical mode.

Reply:
So, if read-only, `Backspace` should have no effect, but `Delete` no matter the value.
if the direction is vertical, both should have no effect.